shmem.c

```diff
@@ -120,11 +123,58 @@ void addOverTimeClientSlot() {
 	}
 }
 
+/// Create a read/write lock
+pthread_rwlock_t create_rwlock() {
+	pthread_rwlockattr_t lock_attr = {};
+	pthread_rwlock_t lock = {};
+
+	// Initialize the lock attributes
+	pthread_rwlockattr_init(&lock_attr);
+
+	// Allow the lock to be used by other processes
+	pthread_rwlockattr_setpshared(&lock_attr, PTHREAD_PROCESS_SHARED);
+
+	// Initialize the lock
+	pthread_rwlock_init(&lock, &lock_attr);
+
+	return lock;
+}
+
+void shm_read_lock() {
+	int result = pthread_rwlock_rdlock(sharedMemoryLock);
+
+	if(result != 0)
+		logg("Failed to obtain SHM read lock: %s", strerror(result));
+}
+
+void shm_write_lock() {
+	int result = pthread_rwlock_wrlock(sharedMemoryLock);
```

Comment:
We should a comment like: The calling thread acquires the write lock if no other thread (regardless if reader or writer) holds the lock. Otherwise, the thread blocks (that is, does not return from `shm_write_lock()`) until it can acquire the lock. Only one thread can be a writer (at the same time) and hold the lock. Any reader can block any writer while readers do not block themselves.

Reply:
See the header for the documentation.